File: enode/src/main/java/org/enodeframework/domain/AggregateSnapshotter.java
package org.enodeframework.domain;

import java.util.concurrent.CompletableFuture;

/**
 * An interface which can restore aggregate from snapshot storage.
 */
public interface AggregateSnapshotter {
    /**
     * Restore the aggregate from snapshot storage.
     */
    <T extends AggregateRoot> CompletableFuture<T> restoreFromSnapshotAsync(Class<T> aggregateRootType, String aggregateRootId);
}

File: enode/src/main/java/org/enodeframework/domain/AggregateSnapshotter.kt
package org.enodeframework.domain

import java.util.concurrent.CompletableFuture

/**
 * An interface which can restore aggregate from snapshot storage.
 */
interface AggregateSnapshotter {
    /**
     * Restore the aggregate from snapshot storage.
     */
    fun <T : AggregateRoot?> restoreFromSnapshotAsync(
        aggregateRootType: Class<T>,
        aggregateRootId: String
    ): CompletableFuture<T>
}
